File: src/main/java/szewek/craftery/cfapi/AddonSearch.java
package szewek.craftery.cfapi;

import org.jetbrains.annotations.Nullable;

import java.util.Date;

public record AddonSearch(
        int id,
        String name,
        AddonAuthor[] authors,
        String summary,
        String websiteUrl,
        String slug,
        long downloadCount,
        double popularityScore,
        long gamePopularityRank,
        AddonAttachment[] attachments,
        AddonSection categorySection,
        AddonFile[] latestFiles
) {
    @Nullable
    public AddonFile latestFile() {
        AddonFile lf = null;
        var ld = new Date(0);
        for (AddonFile f : latestFiles) {
            if (ld.before(f.fileDate())) {
                lf = f;
                ld = f.fileDate();
            }
        }
        return lf;
    }

    @Nullable
    public AddonAttachment defaultAttachment() {
        for (AddonAttachment a : attachments) {
            if (a.isDefault()) {
                return a;
            }
        }
        return null;
    }
}

File: src/main/kotlin/szewek/craftery/cfapi/AddonSearch.kt
package szewek.craftery.cfapi

class AddonSearch(
    val id: Int,
    val name: String,
    val authors: Array<AddonAuthor>,
    val summary: String,
    val websiteUrl: String,
    val slug: String,
    val downloadCount: Long,
    val popularityScore: Double,
    val gamePopularityRank: Long,
    val attachments: Array<AddonAttachment>,
    val categorySection: AddonSection,
    val latestFiles: Array<AddonFile>
)